src/pocketmine/command/defaults/XpCommand.php

```diff
@@ -68,13 +68,34 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
 				throw new InvalidCommandSyntaxException();
 			}else{
 				$level = (int) $check;
-				$player->addXpLevels($level);
-				Command::broadcastCommandMessage($sender, new TranslationContainer("%commands.xp.success.level", [$level, $player->getName()]));
+				if($level < 0){
+					if($player->getXpLevel() + $level < 0){
+						$sender->sendMessage(TextFormat::RED. "Players cannot have negative levels");
+						return true;
+					}
+					
+					$player->subtractXpLevels(-$level);
```

Comment:
Not sure if using subtract here (and in the block below) is a great idea but I just didn't want to put `$player->addXpLevels($level);` at last after everything has done.